internal/broker/bind_create.go

```diff
@@ -104,6 +105,28 @@ func (b *BindEndpoint) Bind(ctx context.Context, instanceID, bindingID string, d
 		}
 	}
 
+	bindingList, err := b.bindingsStorage.ListByInstanceID(instanceID)
+	switch {
+	case dberr.IsNotFound(err):
+		return domain.Binding{}, apiresponses.ErrInstanceDoesNotExist
```

Comment:
According to OSB API we cannot return 404 on the PUT request

Reply:
I will change it to 500 as this may be an internal KEB error even when the user provided valid params.

Reply:
I am not sure if 500 is the best option. Could we differentiate internal error and not existing instance to return either 400 or 500 accordingly?

Reply:
This would require to review the whole bindings code in this PR as we use widely 500. There is a separate subtask for this
https://github.com/kyma-project/kyma-environment-broker/issues/1198#issuecomment-2404880767

Reply:
We decided to leave 500 as it is and differentiate two cases later on in scope of other ticket.